design-library/src/components/BccTooltip/BccTooltip.vue

```diff
@@ -0,0 +1,33 @@
+<script setup lang="ts">
+type Props = {
+  primaryPosition?: "top" | "bottom";
+  secondaryPosition?: "left" | "center" | "right";
+  variant?: "dark" | "white" | "grey";
+  visible?: boolean;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  primaryPosition: "top",
+  secondaryPosition: "center",
+  variant: "dark",
+  visible: false,
+});
+</script>
+
+<template>
+  <div class="bcc-tooltip">
+    <template v-if="$slots.child">
+      <slot name="child"></slot>
+    </template>
+    <div
+      :class="[
+        'bcc-tooltip-content',
+        `bcc-tooltip-${props.primaryPosition}-${props.secondaryPosition}`,
+        `bcc-tooltip-${props.variant}`,
```

Comment:
Interesting that this works. I thought Tailwind purged the classes, and then this wouldn't work as Tailwind has no way to know what class this results in. Their documentation [warns against](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) doing it like this.

It's been some time since I worked on this, but I guess that for Storybook the CSS is just concatenated from all the files and not touched by TW, and that it wouldn't work when using the package, as users will set it up [to purge](https://developer.bcc.no/bcc-design/design-library/vue-components.html#import-tokens-in-your-tailwind-config) unused classes.

That's why in other components we do it like this, so that Tailwind can find the full class name:

```
const sizes = {
  xs: "bcc-button-xs",
  sm: "bcc-button-sm",
  base: "",
  lg: "bcc-button-lg",
  xl: "bcc-button-xl",
};
```